### DISCUSSION TITLE
Ansible Modules

### DISCUSSION THREAD

**Discussion by Thulium-Drake** [semaphoreui/semaphore#1163] [Ideas]
Hi there,

Are there any plans to make modules for Ansible to configure Semaphore with? I'd really like to give it a spin, but I would really like to configure it with Ansible so I have to only do it once :-)

**Comment by fiftin** [comment]
Hi @Thulium-Drake

What this module should to do? Create configuration file for Semaphore?

**Comment by Thulium-Drake** [comment]
Hi @fiftin,

Thanks for the reply! That mostly depends on how Semaphore should be configured (the app itself before you run it, I think that can also be managed with the existing copy/template modules :-) ), but I was mainly talking about modules such as the ```awx.awx``` collection (https://github.com/ansible/awx/tree/devel/awx_collection/plugins/modules), that interact with the REST API of AWX/Tower to create inventories/credentials etc.

From what I've seen so far in the docs, the project is already halfway there, as the REST API is already implemented :-)

**Comment by ansibleguy** [comment]
Greetings

I personally don't see a need for some management actions like setup/config file.
Modules to interact with the Rest-API may make sense.

**Comment by ansibleguy** [comment]
BTW: There seems to be an existing [Ansible Semaphore Collection](https://github.com/semaphoreui/bnc.semaphore) with [some modules](https://github.com/semaphoreui/bnc.semaphore/tree/main/plugins/modules) that allow API interaction

**Comment by Thulium-Drake** [comment]
@fiftin Is that collection @ansibleguy mentioned still maintained? It seems like you own the github org that hosts it :)

And if so, is it compatible with the current version of Semaphore?

**Comment by ansibleguy** [comment]
As the last commit is from 2022 I would guess that project was abandoned.
@Thulium-Drake I have experience (see [ansibleguy.collection_opnsense](https://github.com/ansibleguy/collection_opnsense)) - if you want to contribute to creating one, we could team up.
